opcua.cpp

```diff
@@ -722,18 +722,28 @@ int n_subscriptions = 0;
 /**
  * Stop all subscriptions and disconnect from the OPCUA server
  */
-void
-OPCUA::stop()
+void OPCUA::stop()
 {
-	if (m_connected)
+	try
 	{
-		subscriptionVariables.clear();
-		m_assetPathNames.clear();
-		m_client->Disconnect();
+		if (m_connected)
+		{
+			subscriptionVariables.clear();
+			m_assetPathNames.clear();
+			m_client->Disconnect();			
```

Comment:
put this inside an if (m_client) test

Reply:
The following code is executed under the `m_connected` flag, which guarantees that `m_client` is never null at this point. However, it’s important to note that `m_client` (of type `UaClient`) is not exception-safe. The `Disconnect()` method can throw exceptions, and since `Disconnect()` is called in `delete m_client`, this raises a risk of memory leaks if an exception occurs during disconnection.

Currently, our implementation ensures that the process exits after invoking `OPCUA::stop()`, which protects us from memory leaks under normal circumstances.

If you agree i would like to add this finding as comments here ..